src/controller/proxy/remote.go

```diff
@@ -43,13 +44,16 @@ type remoteHelper struct {
 	regID       int64
 	registry    adapter.ArtifactRegistry
 	registryMgr reg.Manager
+	opts        *Options
 }
 
 // NewRemoteHelper create a remote interface
-func NewRemoteHelper(ctx context.Context, regID int64) (RemoteInterface, error) {
+func NewRemoteHelper(ctx context.Context, regID int64, speed int32) (RemoteInterface, error) {
```

Comment:
The use of option closure is for subsequent better scalability, if speed is passed directly as a function parameter then it loses the significance of the option method, a more reasonable use of the following:

```go
func NewRemoteHelper(ctx context.Context, regID int64, opts ...Option) (RemoteInterface, error) {
    o := NewOptions(opts)
}

// call NewRemoteHelper
_, _  := proxy.NewRemoteHelper(ctx, regID, WithSpeed(speed))
```

Reply:
Good idea!